includes/tracks/events/class-wc-coupons-tracking.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * WooCommerce Coupons Tracking
+ *
+ * @package WooCommerce\Tracks
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * This class adds actions to track usage of WooCommerce Orders.
+ */
+class WC_Coupons_Tracking {
+	/**
+	 * Init tracking.
+	 */
+	public function init() {
+		add_action( 'load-edit.php', array( $this, 'tracks_coupons_events' ), 10 );
+	}
+
+	/**
+	 * Add a listener on the "Apply" button to track bulk actions.
+	 */
+	public function tracks_coupons_bulk_actions() {
+		wc_enqueue_js(
+			"
+			var apply = $( '#posts-filter input[type=\"submit\"]' );
+			$( apply ).on( 'click', function() {
+				var action = $('select[name=\"action\"]').val();
+				if ( '-1' !== action ) {
+					window.wcTracks.recordEvent( 'coupons_view_bulk_action', {
+						action: action
+					} );
+				}
+			} );
+		"
+		);
+	}
+
+	/**
+	 * Track page view events.
+	 */
+	public function tracks_coupons_events() {
+		if ( isset( $_GET['post_type'] ) && 'shop_coupon' === wp_verify_nonce( sanitize_key( $_GET['post_type'] ) ) ) {
```

Comment:
I believe there is an error in the logic here and this if will never be true. Maybe we should check `'shop_coupon' === $_GET['post_type']` instead of `'shop_coupon' === wp_verify_nonce( sanitize_key( $_GET['post_type'] ) )`? Also, I don't think we want to pass `$_GET['post_type']` as the first param to `wp_verify_nonce()` as this variable doesn't contain a nonce value.

Reply:
Yes, good spotting. That was definitely happening. Fixed in https://github.com/woocommerce/woocommerce/pull/23189/commits/bc233409096c2693d92892a6d216216844b4d2b9